configuration/yaml.md

```diff
@@ -0,0 +1,660 @@
+| `<channel_id>`            | The channel ID. It is a single word that when combined with the Thing UID forms the full channel UID.                                                                                |
+| `type`                    | Channel type as defined by the Binding. When the channel type is defined, `kind`, `itemType`, and `itemDimension` should not be defined because the channel type will override them. |
+| `kind`                    | Whether it's a State channel or a Trigger channel. Default: `state`                                                                                                                  |
+| `itemType`                | The item type that this channel can be linked to. Usually this is defined by the binding when channel `type` is defined.                                                             |
+| `itemDimension`           | This may be used to further define a `Number` `itemType`.                                                                                                                            |
+| `label`                   | Channel label.                                                                                                                                                                       |
+| `description`             | Channel description.                                                                                                                                                                 |
+| `config`                  | A key-value map of the channel's configuration. Refer to the Binding's documentation for details.                                                                                    |
+
+Example:
+
+```yaml
+version: 1
+
+things:
+  mqtt:broker:mosquitto:
+    isBridge: true
+    config:
+      host: 192.168.1.1
+      lwtMessage: offline
+      lwtTopic: openhab/status
+      lwtQos: 1
+      birthMessage: online
+      birthTopic: openhab/status
+      shutdownMessage: offline
+      shutdownTopic: openhab/status
+      enableDiscovery: false
+
+  mqtt:topic:bedroom1-switch:
+    bridge: mqtt:broker:mosquitto
+    label: Bed Room1 Switch
+    config:
+      availabilityTopic: bedroom1-switch/status
+      payloadAvailable: online
+      payloadNotAvailable: offline
+    channels:
+      switch1:
+        type: switch
+        config:
+          stateTopic: bedroom1-switch/switch/switch1/state
+          commandTopic: bedroom1-switch/switch/switch1/command
+      button1:
+        type: string
+        config:
+          stateTopic: bedroom1-switch/button1
+          postCommand: true
+      statusled:
+        type: switch
+        config:
+          stateTopic: bedroom1-switch/light/status_led/state
+          commandTopic: bedroom1-switch/light/status_led/command
+          transformationPattern:
+            - JSONPATH:$.state
+          formatBeforePublish: '{"state": "%s"}' # This needs to be quoted to be parsed as a string
```

Comment:
quoted with either ' or "

Reply:
It's a bit more complicated, and I don't want to be too verbose here. If it's quoted using double quotes, the double quotes inside needs to be escaped. So just saying `either ' or "` may lead to more problems. 

I expect people to just intuitively know what to use, or read up on a more detailed explanation on yaml. Therefore I think I'll leave this comment as is.